unstoppable-ios-app/domains-manager-ios/Services/WalletConnectService_V2/WalletConnectServiceV2.swift

```diff
@@ -381,7 +404,33 @@ class WalletConnectServiceV2: WalletConnectServiceV2Protocol {
     }
 }
 
+extension WalletConnectServiceV2: DataAggregatorServiceListener {
+    func dataAggregatedWith(result: DataAggregationResult) {
+        if case .success(let serviceResult) = result,
+           case .domainsUpdated(let validDomains) = serviceResult {
+            
+            let unifiedApps = getAllUnifiedAppsFromCache()
+            let validConnectedApps = unifiedApps.trimmed(to: validDomains)
+            
+            disconnectApps(from: unifiedApps, notIncluding: validConnectedApps)
```

Comment:
Duplicated logic of `getConnectedApps()` ?

Reply:
They share the logic but I wouldn't mix the method that fetches smth with a method that does smth silently